corehq/ex-submodules/casexml/apps/phone/models.py

```diff
@@ -354,6 +356,94 @@ def test_only_get_dependent_cases_on_phone(self):
         raise NotImplementedError()
 
 
+def save_synclog_to_sql(synclog_json_object):
+    synclog = synclog_to_sql_object(synclog_json_object)
+    synclog.save()
+
+
+def delete_synclog(synclog_id):
+    """
+    Deletes synclog object both from Couch and SQL, if the doc
+        isn't found in either couch or SQL an exception is raised
+    """
+    deleted = False
+    try:
+        synclog = SyncLogSQL.objects.filter(synclog_id=synclog_id).first()
+        if synclog:
+            synclog.delete()
+            deleted = True
+    except ValidationError:
+        pass
+
+    try:
+        synclog = properly_wrap_sync_log(SyncLog.get_db().get(synclog_id))
+        SyncLog.get_db().delete_doc(synclog)
+        deleted = True
+    except ResourceNotFound:
+        pass
+
+    if not deleted:
+        raise MissingSyncLog("A SyncLogSQL object with this synclog_id ({})is not found".format(synclog_id))
+
+
+def synclog_to_sql_object(synclog_json_object):
+    # Returns a SyncLogSQL object, a saved instance from DB or
+    #   instantiated SQL object to be saved
+    # synclog_json_object should be a SyncLog instance
+    # if synclog_json_object._id
+    synclog = None
+    if synclog_json_object._id:
+        synclog = SyncLogSQL.objects.filter(synclog_id=synclog_json_object._id).first()
+
+    is_new_synclog_sql = not synclog_json_object._id or not synclog
+
+    if is_new_synclog_sql:
+        synclog_id = synclog_json_object._id or uuid.uuid1().hex.lower()
+        synclog_json_object._id = synclog_id
+        synclog = SyncLogSQL(
+            domain=synclog_json_object.domain,
+            user_id=synclog_json_object.user_id,
+            synclog_id=synclog_id,
+            date=synclog_json_object.date,
+            previous_synclog_id=getattr(synclog_json_object, 'previous_log_id', None),
+            log_format=synclog_json_object.log_format,
+            build_id=synclog_json_object.build_id,
+            duration=synclog_json_object.duration,
+            last_submitted=synclog_json_object.last_submitted,
+            had_state_error=synclog_json_object.had_state_error,
+            error_date=synclog_json_object.error_date,
+            error_hash=synclog_json_object.error_hash,
+        )
+    synclog.doc = synclog_json_object.to_json()
+    return synclog
+
+
+@architect.install('partition', type='range', subtype='date', constraint='day', column='date')
```

Comment:
If we're keeping these around indefinitely I think we should change this constraint to either week or month. 

I just ran into an issue of poor insert performance while testing one of the new UCRs and will be changing that to month. Good rule of thumb is ~100 partitions according to [docs](https://www.postgresql.org/docs/10/static/ddl-partitioning.html)

month gives us ~8 years of data before it should impact performance, weeks gives ~2 years. If it's ever ok to prune this to < 3 months, then day should be good

Reply:
Hey, so per Simon's below comment we are gonna keep this to day, right?

Reply:
@sravfeyn I think he meant to change it to week

Reply:
yes, change to week

Reply:
Updated